src/main/java/it/gov/pagopa/pu/workflow/event/payments/consumer/PaymentsConsumer.java

```diff
@@ -43,6 +46,15 @@ public void accept(PaymentEventDTO paymentEventDTO) {
                   .transferIndex(t.getTransferIndex())
                   .build()
               )));
+
+        Long receiptIdFromEvent= Long.valueOf(paymentEventDTO.getEventDescription().replace("receiptId:",""));
+        debtPosition.getPaymentOptions().stream()
+          .flatMap(paymentOptionDTO -> paymentOptionDTO.getInstallments().stream())
+          .filter(installment -> receiptIdFromEvent.equals(installment.getReceiptId()))
+          .findFirst()
+          .ifPresent(installment -> createAssessmentsWFClient.createAssessments(receiptIdFromEvent));
```

Comment:
move this on a separate method, let's try/catch the receiptId retrieve.
if the it throw an exception, let's consider all receiptId on the provided Installment
```suggestion
        List<Long> receiptIds;
        try{receiptIds= List.of(Long.valueOf(paymentEventDTO.getEventDescription().replace("receiptId:","")));
        }catch(Exception e){
        log.error("It was not possible to retrieve the particular receiptId which originated the event, let's considering all installment's receiptIds");
        receiptIds = debtPosition.getPaymentOptions().stream()
                                   .flatMap(paymentOptionDTO -> paymentOptionDTO.getInstallments().stream())
                                   .map(InstallmentDTO::getReceiptId)
                                    .filter(Object::nonNull)
                                    .toList();
        }
        if(receiptIds.isEmpty()){
          log.error("Cannot retrieve a receiptId related to the input event:" + paymentEventDTO.getEventId());
        }
        receiptIds.forEach(receiptId -> createAssessmentsWFClient.createAssessments(receiptId));
```